src/main/java/orionmipt/Context.java

```diff
@@ -0,0 +1,309 @@
+package orionmipt;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+
+class Context implements Cloneable {
+
+    private final Map<Object, String> classAttributeMap;
+    private final Map<Object, String> fieldAttributeMap;
+    private final Set<Object> pretendPrimitiveSet;
+    private final Set<Object> ignoreSet;
+
+    /**
+     * Allow private, protected and package-access fields to be shown.
+     * This is only possible if the security manager allows
+     * <code>ReflectPermission("suppressAccessChecks")</code> permission.
+     * This is usually the case when running from an application, but
+     * not from an applet or servlet.
+     */
+    private boolean ignorePrivateFields = false;
```

Comment:
Here's an idea for you: instead of defining a number of boolean fields here, define an `enum` with all the boolean features, and use a mutable `EnumSet` to hold all the flags.

Thus you will get a class with all the fields marked final, but with mutable state, which is convenient for building.